src/trace/context.ts

```diff
@@ -152,7 +195,7 @@ export function readTraceFromEvent(event: any): TraceContext | undefined {
 
   // e.g. When lambda is invoked synchronously, headers can be set to null by the caller
   if (!headers || typeof headers !== "object") {
-    return;
+    return readTraceFromSQSEvent(event);
```

Comment:
Do you mind helping refactor the rest of the logic to something like `reaadTraceFromHTTPEvent`? So we keep the code flow structure like this?

- readTraceFromEvent
  - readTraceFromSQSEvent
  - reaadTraceFromHTTPEvent

Reply:
Done - please re-review @tianchu